test/sdf_tape_test.cpp

```diff
@@ -0,0 +1,201 @@
+// Copyright 2024 The Manifold Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <chrono>
+#include <iostream>
+
+#include "../src/sdf/tape.h"
+#include "../src/sdf/value.h"
+#include "test.h"
+
+using namespace manifold;
+using namespace manifold::sdf;
+
+int recursive_interval(EvalContext<Interval<double>>& ctx, vec3 start,
```

Comment:
@elalish change `NearSurface` into something like this that is recursive and takes a function that outputs an interval.